new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,438 @@ description: Learn about writing Oso policies - the source of truth for authoriz
+matches a rule if:
 
-```polar
-class User:
-    ...
+* the names match, _and_
+* each query argument matches the corresponding rule parameter, _and_
+* queries for each term in the rule body all succeed.
 
-class Admin(User):
-    ...
-```
+(Note that the last condition is trivially true if the rule
+has no body.)
 
-…we can modify our original bodiless rule to only allow `Admin` users to
-approve any expense report by adding a
-[specializer](#registering-application-types) to the rule
-head:
+Let's make this concrete by considering a very simple policy:
+suppose users named Abagail, Carol, and Johann are allowed to
+read some resource. We'll represent users, actions, and resources
+as strings for now, but we'll see richer representations in just
+a moment. We could express this policy with the following three
+Polar rule definitions:
 
 ```polar
-allow(_actor: Admin, "approve", _report);
+allow("Abagail", "read", "resource");
+allow("Carol", "read", "resource");
+allow("Johann", "read", "resource");
 ```
 
-The rule will fail when evaluated on a regular `User` and succeed when
-evaluated on an `Admin`, encoding an implicit `if Admin` condition.
+These rules represent _facts_ — statements that are unconditionally
+true. After we load this file into the knowledge base, Oso "knows"
+these three facts.
 
-This is another example of the rule matching process: instead of matching
-against a concrete value, we are instead checking to make sure the type of the
-input matches the expected type — in this case, an `Admin`.
+We can then make authorization decisions from our {{< lang >}}
+application by calling:
 
-{{% callout "Tip" "green" %}}
-  Try to use type specializers as often as possible. It will help make sure you
-  don't accidentally allow access to an unrelated resource which happens to
-  have matching fields.
-{{% /callout %}}
+```python
+oso.is_allowed("Johann", "read", "resource")
+```
 
-### Combining Rules
+The Oso library issues a Polar query to the knowledge base:
 
-Rules can be thought of as equivalent to methods in imperative programming. The
-same idea should be applied when writing policies: any piece of logic that you
-want to reuse throughout a policy can be extracted out into a new rule.
+```polar
+allow("Johann", "read", "resource")
+```
 
-The benefits of that extraction are that it makes it easier to keep logic
-consistent throughout and often results in a much more readable policy.
+The knowledge base _searches_ for a definition that matches the query.
+In this case all three rules match the name `allow`, but the
+first two rules' first parameters fail to match the first supplied
+argument, since (in Polar notation), `"Abagail" != "Johann"` and
+`"Carol" != "Johann"`. The third rule, however, matches each of
+the arguments with its corresponding parameter: `"Johann" = "Johann"`,
+`"read" = "read"`, and `"resource" = "resource"`. So the query
+succeeds, and `oso.is_allowed` returns `True`.
```

Comment:
I think we need one sentence here explaining that only one rule must succeed for a query to succeed, to make it clear that rules are only additive and not subtractive. We could also state it as each rule definition represents an implicit "or". Just something to hit that home at the end of this.